### PR TITLE
Add is_native and fe_native functions

### PR DESCRIPTION
**Pull request by ibartomeus**

### CODE REVIEW

File: R/fe_native.R [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                                 "Hungary", "Iceland", "Italy", "Jugoslavia",
+                                 "Portugal", "Norway", "Poland", "Romania",
+                                 "USSR", "Sardegna", "Svalbard", "Sicilia",
+                                 "Sweden", "Turkey", "USSR_Northern_Division",
+                                 "USSR_Baltic_Division", "USSR_Central_Division",
+                                 "USSR_South_western", "USSR_Krym",
+                                 "USSRSouth_eastern_Division"))
+  require(plyr)
+  require(XML)
+  require(RCurl)
```
Comment (sckott) [inline R/fe_native.R]: @ibartomeus don't do require/library calls in the function itself. can you remove these 3 lines?  

also, doesn't look like you use plyr here

File: R/is_native.R [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#' sp <- c("Lavandula stoechas", "Carpobrotus edulis", "Rhododendron ponticum",
+#'        "Alkanna lutea", "Anchusa arvensis")
+#' is_native(sp[1], where = "Islas_Baleares", region = "europe")
+#' sapply(sp, is_native, where = "Continental US", region = "america")
+#' sapply(sp, is_native, where = "Islas_Baleares", region = "europe")
+#' }
+#'
+#'
+is_native <- function(sp, where, region = c("america", "europe")){
+  require(taxize)
```
Comment (sckott) [inline R/is_native.R]: @ibartomeus  again, do remove these library load calls in the fxn

File: R/fe_native.R [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  genus <- strsplit(sp, " ")[[1]][1]
+  species <- strsplit(sp, " ")[[1]][2]
+  #create urls to parse
+  urls <- paste("http://rbg-web2.rbge.org.uk/cgi-bin/nph-readbtree.pl/feout?FAMILY_XREF=&GENUS_XREF=",
+                  genus, "&SPECIES_XREF=", species, "&TAXON_NAME_XREF=&RANK=",
+                  sep = "")
+  print(paste("Checking species", sp))
+  #Parse url and extract table
+  #readHTMLTable(urls) #Not working, don't know why.
+  doc <- htmlTreeParse(urls, useInternalNodes = TRUE)
```
Comment (sckott) [inline R/fe_native.R]: @ibartomeus Can you use `httr::GET` and check for status code here first, then use `htmlTreeParse` ? so we can make sure to give users more useful information when a request fails

File: not_used_files/colsearch2.R [hunk 4]
```diff
@@ -0,0 +1,67 @@
+name = "Carpobrotus edulis"
```
Comment (sckott) [inline not_used_files/colsearch2.R]: @ibartomeus can you delete this file, or do you need it?

File: not_used_files/usda_native.R [hunk 5]
```diff
@@ -0,0 +1,33 @@
+sp = "Abutilon incanum"
```
Comment (sckott) [inline not_used_files/usda_native.R]: @ibartomeus can you delete this file, or do you need it?

Comment (ibartomeus) [inline not_used_files/usda_native.R]: Done!

(I think, I was not sure about the use of GET) I removed library call's
(but do I need to add those somewhere else?)

Should I need to PR again?

Thanks for the comments
Nacho

On Fri, May 8, 2015 at 6:06 PM, Scott Chamberlain <notifications@github.com>
wrote:

> In not_used_files/usda_native.R
> <https://github.com/ropensci/traits/pull/28#discussion_r29952402>:
>
> > @@ -0,0 +1,33 @@
> > +sp = "Abutilon incanum"
>
> @ibartomeus <https://github.com/ibartomeus> can you delete this file, or
> do you need it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ropensci/traits/pull/28/files#r29952402>.
>



-- 
Ignasi Bartomeus PhD
www.bartomeuslab.com
@ibartomeus
Skype: nachobartomeus

Dpto. Ecología Integrativa
Estación Biológica de Doñana (EBD-CSIC)
Avda. Américo Vespucio s/n
Isla de la Cartuja
41092, Sevilla (Spain)

**Comment by sckott** [comment]
thanks @ibartomeus let me know when you're done

**Comment by ibartomeus** [comment]
I am done.

**Comment by sckott** [comment]
great, thanks

**Comment by sckott** [comment]
@ibartomeus looks good, i have a few small internal changes I'd like to make after merging if that's okay

**Comment by ibartomeus** [comment]
Please change whatever you need. I am aware code can be enhanced.

On Tue, May 12, 2015 at 12:24 AM, Scott Chamberlain <
notifications@github.com> wrote:

> Merged #28 <https://github.com/ropensci/traits/pull/28>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ropensci/traits/pull/28#event-302246784>.
>



-- 
Ignasi Bartomeus PhD
www.bartomeuslab.com
@ibartomeus
Skype: nachobartomeus

Dpto. Ecología Integrativa
Estación Biológica de Doñana (EBD-CSIC)
Avda. Américo Vespucio s/n
Isla de la Cartuja
41092, Sevilla (Spain)

**Comment by sckott** [comment]
thanks @ibartomeus :)

**Pull request merged**